pkg/objstore/objstore.go

```diff
@@ -172,29 +232,43 @@ func NopCloserWithSize(r io.Reader) io.ReadCloser {
 
 // UploadDir uploads all files in srcdir to the bucket with into a top-level directory
 // named dstdir. It is a caller responsibility to clean partial upload in case of failure.
-func UploadDir(ctx context.Context, logger log.Logger, bkt Bucket, srcdir, dstdir string) error {
+func UploadDir(ctx context.Context, logger log.Logger, bkt Bucket, srcdir, dstdir string, options ...UploadOption) error {
 	df, err := os.Stat(srcdir)
+	opts := applyUploadOptions(options...)
+	g, ctx := errgroup.WithContext(ctx)
+	g.SetLimit(opts.concurrency)
```

Comment:
SetLimit was introduced fairly recently: https://go-review.googlesource.com/c/sync/+/405174

So needed to update the `golang.org/x/sync` package.